src/metadata.ts

```diff
@@ -13,11 +14,14 @@ export type Metadata = {
  * @param encodedName will be coerced to string, so… whatever
  */
 export function normalizeMetadata(input?: File | Response | BufferLike | StreamLike, encodedName?: any, size?: number | bigint): Metadata {
-  if (encodedName !== undefined && !(encodedName instanceof Uint8Array)) encodedName = encodeString(encodedName)
+  const isEncodedNameUtf8 = !(encodedName instanceof Uint8Array )
+  
+  if (encodedName !== undefined && isEncodedNameUtf8) encodedName = encodeString(encodedName)
```

Comment:
The mistake was mine to begin with but I just noticed : if the filename is a raw ArrayBuffer (which could be the result of client-side decryption with the WebCrypto API, for example) or some kind of ArrayView but not a Uint8Array, we're going to coerce it to a String and encode it as UTF-8. That will fail miserably, of course. In all those cases we should create a Uint8Array directly from the ArrayBuffer.